### PR TITLE
Added logging events to dataStorage mock

### PR DESCRIPTION
**Pull request by hAdamm**

### CODE REVIEW

File: tests/__assets/DataStorage/DataStorage.php [hunk 1]
```diff
@@ -4,8 +4,40 @@
...
+    protected $eventsStack = [
+        "inserts" => ["all" => [], "byEntities" => []],
+        "updates" => ["all" => [], "byEntities" => []],
+        "deletes" => ["all" => [], "byEntities" => []],
+        ];
+
+    protected function logEventForEntity($data, $eventType, \obo\Entity $entity) {
+        $this->eventsStack[$eventType]["all"][] = ["type" => $eventType, "data" => $data, "entity" => $entity];
+        $this->eventsStack[$eventType]["byEntities"][$entity->objectIdentificationKey()][]
+                = ["type" => $eventType, "data" => $data, "entity" => $entity];
```
Comment (ZZromanZZ) [inline tests/__assets/DataStorage/DataStorage.php]: is new line on purpose ?

File: tests/__assets/DataStorage/DataStorage.php [hunk 2]
```diff
@@ -4,8 +4,40 @@
...
+    const EVENT_TYPE_UPDATE = "updates";
+    const EVENT_TYPE_DELETE = "deletes";
+
     public static $autoIncrementIndex = 0;
 
+    protected $eventsStack = [
+        "inserts" => ["all" => [], "byEntities" => []],
+        "updates" => ["all" => [], "byEntities" => []],
+        "deletes" => ["all" => [], "byEntities" => []],
+        ];
```
Comment (ZZromanZZ) [inline tests/__assets/DataStorage/DataStorage.php]: indetation

File: tests/__assets/DataStorage/DataStorage.php (hunk 1 above)
Comment (ZZromanZZ) [inline tests/__assets/DataStorage/DataStorage.php]: Isn't this better?
```php
$event = ["type" => $eventType, "data" => $data, "entity" => $entity];
$this->eventsStack[$eventType]["all"][] = $event;
$this->eventsStack[$eventType]["byEntities"][$entity->objectIdentificationKey()][] = $event;
```

Comment (hAdamm) [inline tests/__assets/DataStorage/DataStorage.php]: Yes, definitely yes

**Pull request merged**